mlflow/types/llm.py

```diff
@@ -292,18 +323,8 @@ class ChatChoice(_BaseDataclass):
     def __post_init__(self):
         self._validate_field("index", int, True)
         self._validate_field("finish_reason", str, True)
-        if isinstance(self.message, dict):
-            self.message = ChatMessage(**self.message)
-        if not isinstance(self.message, ChatMessage):
-            raise ValueError(
-                f"Expected `message` to be of type ChatMessage or dict, got {type(self.message)}"
-            )
-        if isinstance(self.logprobs, dict):
-            self.logprobs = ChatChoiceLogProbs(**self.logprobs)
-        if self.logprobs and not isinstance(self.logprobs, ChatChoiceLogProbs):
-            raise ValueError(
-                f"Expected `logprobs` to be of type LogProbs or dict, got {type(self.logprobs)}"
-            )
+        self._convert_dataclass("message", ChatMessage, True)
+        self._convert_dataclass("logprobs", ChatChoiceLogProbs, False)
```

Comment:
I guess maybe a better way is to make `from_dict` call field value `from_dict` recursively(in this way we need to define `from_dict` for each class).  So that we don't need to inject `_convert_dataclass` invocation into `__post_init__`

Reply:
Can be a follow-up work